pallets/compliance-manager/src/lib.rs

```diff
@@ -735,9 +843,22 @@ impl<T: Trait> Module<T> {
                 asset_compliance_with_results.result = true;
             }
         }
+        // Aggregating the result with the implicit requirement result.
+        asset_compliance_with_results.result =
+            Self::compute_final_result(&asset_compliance_with_results);
         asset_compliance_with_results
     }
 
+    fn compute_final_result(asset_compliance: &AssetComplianceResult) -> bool {
+        let compliance_result = asset_compliance.result;
```

Comment:
`result` appears to be used intermediately as opposed to what the documentation in the struct which says it stored the final result. `compute_final_result` literally computes the final result given a final result as an argument, which doesn't sound right.

It's probably best to fix the confusion at this stage when you are actively using the `result` field rather than carry on with using the field in undocumented ways.

Reply:
I don't think a confusion here. we are still storing the final result to `result` variable. We only used result to calculate the final result.